### PR TITLE
c-blosc: Add -std=gnu99 flag when gcc.

### PR DESCRIPTION
**Pull request by tkameyama**
c-blosc add -std=gnu99 flag if x86_64 and gcc.
But -std=gnu99 flag is needed for all target.
This fix is added -std=gnu99 flag when compiler is gcc.

### CODE REVIEW

File: var/spack/repos/builtin/packages/c-blosc/package.py [hunk 1]
```diff
@@ -32,6 +32,8 @@ class CBlosc(CMakePackage):
     depends_on('zstd')
     depends_on('lz4')
 
+    patch('gcc.patch')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/c-blosc/package.py]: @tkameyama will this patch work on versions other than `1.16.3`? If not, could you add `when='@1.16.3'`?

Comment (tkameyama) [reply var/spack/repos/builtin/packages/c-blosc/package.py]: I'm checked version now. And add when=.

**Comment by tkameyama** [comment]
Since #12110 was merged, I rebased

**Comment by tkameyama** [comment]
Because #12204 is merged, I rebased

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**